articles/quickstart/webapp/golang/interactive.md

```diff
@@ -0,0 +1,172 @@
+
+## Configure OAuth2 and OpenID Connect packages {{{ data-action=code data-code="auth.go" }}}
+
+In order to be able to use Auth0, the OAuth2 and OpenID Connect packages need to be configured. 
+
+Create a file called `auth.go` in the `platform/authenticator` folder. In this package, create a method to 
+configure and return [OAuth2](https://godoc.org/golang.org/x/oauth2) and 
+[OIDC](https://godoc.org/github.com/coreos/go-oidc) clients, and another one to verify an ID Token.
+
+## Set up your application routes {{{ data-action=code data-code="router.go" }}}
+
+Create a file called `router.go` in the `platform/router` folder. In this package, create a method to configure
+and return our routes using [github.com/gin-gonic/gin](https://github.com/gin-gonic/gin). You will be passing an
+instance of `Authenticator` to the method, so it can be used within the `login` and `callback` handlers.
+
+::: note
+The router uses the [github.com/gin-contrib/sessions](https://github.com/gin-contrib/sessions) middleware to manage our cookie based sessions.
+:::
+
+## Serve your application {{{ data-action=code data-code="main.go" }}}
+
+With both the authenticator and router configured, we can wire things up using our
+application's entry point. Inside `main.go`, create an instance of the authenticator and the router, which gets passed the authenticator instance.
+
+If you are using a `.env` file, like we are in this example, be sure to call `godotenv.Load()` at the very beginning of the `main()` function.
+
+## Add login to your application {{{ data-action=code data-code="login.go" }}}
+
+In order for user to be able to authenticate themselves, we need to create a handler function to handle the `/login` route.
+
+Create a file called `login.go` in the `web/app/login` folder, and add a `Handler` function. Upon executing the handler, the user will be redirect to Auth0 where they can enter their credentials.
+
+In order to be able to call the `/login` route, add a link to `/login` in the `home.html` template located in the `web/template` directory.
+
+```html
+<!-- web/template/home.html -->
+<div>
+    <h3>Auth0 Example</h3>
+    <p>Zero friction identity infrastructure, built for developers</p>
+    <a href="/login">SignIn</a>
+</div>
+```
+
+## Handle authentication callback {{{ data-action=code data-code="callback.go" }}}
+
+Once users have authenticated using Auth0's Universal Login Page, they'll return to the app at the `/callback` route.
+
+Create a file called `callback.go` in the `web/app/callback` folder, and add a `Handler` function.
+
+This handler will take the `code` querystring, provided by Auth0, and exchange it for an Id Token and an Access Token.
+
+If the Id token is valid, it will store the profile information (which gets build based on the claims in the Id token) in the session along with the access token, this is done in order to be able to retrieve and use that information when needed.
```

Comment:
```suggestion
This handler will take the `code` query string, provided by Auth0, and exchange it for an ID token and an access token.

If the ID token is valid, it will store the profile information and access token in the session. The profile information is based on the claims contained in the ID token. Session storage allows the application to access that information as needed.
```